features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -876,6 +906,23 @@ void GenericSecurityManager::on_valid_mic_timeout(connection_handle_t connection
     (void)connection;
 }
 
+void GenericSecurityManager::on_signature_verification_failure(connection_handle_t connection) {
+    ControlBlock_t *cb = get_control_block(connection);
+    if (!cb) {
+        return;
+    }
+
+    cb->mic_failures++;
+    if (cb->mic_failures == 3) {
```

Comment:
this is the same limit as the what the spec suggests for number of MIC failures before refreshing the keys